### DISCUSSION TITLE
QSO-date and -time differs from lotw-last-upload

### DISCUSSION THREAD

**Discussion by dg9vh** [wavelog/wavelog#130] [Q&A] [answered]
**Describe the bug**

QSO-date and -time differ unplausibly from lotw-last-upload-timestamp of callsign, see screenshot:

![image](https://github.com/wavelog/wavelog/assets/13950650/16adcc57-b772-4367-99df-2992268e978b)

Maybe an timezone-issue of the timestamp? Do we need here to convert the used timezone on the ARRL-system to UTC?

**Comment by dg9vh** [comment]
To be a bit clearer: Here in this case, the QSO-partner confirmed a QSO visually on a time that was before the QSO happend :-)

**Reply by int2001** [reply]
GM Kim.
Misunderstanding. The Timestamp of the green "L" has nothing todo with your particular QSO. It doesn't show when (or if) your QSO was confirmed. It is just showing that the other Station is using LotW and his last upload.

That Data isn't generated out of your own QSOs, it's a service from LotW where Wavelog downloads that data and displays it. Usecase: You can see (before/during/after the QSO), if the User is active on LotW.

Doesn't mean that he confirmed your QSO (or others) at that time. On-Top: That data is sometimes a little outdated (ARRL-Problems). 

HTH?

**Comment by dg9vh** [comment]
It is not a thing of conversion - I also have entries that differs more than 3 days into the past from the QSO-date... It looks like, that the last-update-timestamp from LOTW is not correctly updated when updating LOTW-data (via cron or manually). It seems that the state is saved at the moment, the QSO is logged and this data for this callsign is fetched for the first time.

**Comment by HB9HIL** [comment, marked as answer]
The "Last Upload" data provided by LotW is not what we would call "Live". 

It's a CSV which is generated every few day's and served by LotW. So it can happen, that the User uploaded his Log, you got the confirmation but the `Last Upload` data is not updated. 

Unfortunately we can't solve that in Wavelog.

[https://lotw.arrl.org/lotw-help/developer-query-lotw-users/](https://lotw.arrl.org/lotw-help/developer-query-lotw-users/)

<img width="410" alt="lotw" src="https://github.com/wavelog/wavelog/assets/80885850/306c370c-431f-4b74-91ee-6f9d271e3475">

**Comment by HB9HIL** [comment]
I transfered this issue to a discussion in Q&A

maybe other users have this question aswell :)
